### PR TITLE
fix(ses): align with XS property censorship agreement

### LINKED ISSUE #910: {[inert function]} from toString() for Math.random etc.
In a discussion of API evolution, @erights noted that for new features, testing for new properties by name works well, but for retroactive changes such as making `Math.random()` inert, it's unrealistic to expect code to test for absence of a property.

 - [ ] code changes, unit tests as usual
 - [ ] 262 test

_IOU more explanation, but I have a meeting..._

cc @phoddie

### PR DESCRIPTION
**Pull request by erights**
Fixes #910 

@phoddie, I seem unable to add you as an official reviewer, but please consider yourself one anyway. I'm eager for your comments.

Technically this is a compat break, which I currently mark with a `fix(ses)!:`. Reviewers, please let me know whether you consider this change a compat break, or whether I should demote it to `fix(ses):`. Thanks.

### CODE REVIEW

File: packages/ses/src/tame-date-constructor.js [hunk 1]
```diff
@@ -17,26 +17,38 @@ export default function tameDateConstructor(dateTaming = 'safe') {
 
   // Use concise methods to obtain named functions without constructors.
   const tamedMethods = {
+    /**
+     * `%SharedDate%.now()` throw a `TypeError` starting with "secure mode".
+     * See https://github.com/endojs/endo/issues/910#issuecomment-1581855420
+     */
     now() {
-      return NaN;
+      throw TypeError('secure mode Calling %SharedDate%.now() throws');
```
Comment (kriskowal) [inline packages/ses/src/tame-date-constructor.js:25]: This is an improvement.

File: packages/ses/src/tame-date-constructor.js [hunk 2]
```diff
@@ -17,26 +17,38 @@ export default function tameDateConstructor(dateTaming = 'safe') {
...
+   *     'Invalid Date'
+   *
+   * OriginalDate (normal standard) behavior preserved by
+   * `%InitialDate%`.
+   *   * `Date(anything)` gives a string with the current time
+   *   * `new Date()` returns the current time, as a Date object
+   *
+   * `%SharedDate%` behavior
+   *   * `Date(anything)` throws a TypeError starting with "secure mode"
+   *   * `new Date()` throws a TypeError starting with "secure mode"
```
Comment (kriskowal) [inline packages/ses/src/tame-date-constructor.js:48]: Seems like a missed opportunity.
- Error: No time
- Error: No time for you
- Error: Too late
- Error: Time is an illusion
- Error: Time is a flat circle
- Error: Time flies like an arrow, fruit flies like a banana

File: packages/ses/src/tame-date-constructor.js [hunk 3]
```diff
@@ -69,13 +85,13 @@ export default function tameDateConstructor(dateTaming = 'safe') {
         configurable: false,
       },
       parse: {
-        value: Date.parse,
+        value: OriginalDate.parse,
```
Comment (kriskowal) [inline packages/ses/src/tame-date-constructor.js:88]: There is no original date. The day after New Year’s Eve, 1 BCE was New Year’s Day, 1 CE.

File: packages/ses/src/permits.js [hunk 4]
```diff
@@ -649,12 +649,16 @@ export const permitted = {
...
   '%InitialMath%': {
-    ...SharedMath,
-    // random is standard but omitted from SharedMath
+    ...CommonMath,
+    // `%InitialMath%.random()` has the standard unsafe behavior
     random: fn,
   },
 
-  '%SharedMath%': SharedMath,
+  '%SharedMath%': {
```
Comment (kriskowal) [inline packages/ses/src/permits.js:657]: Or, as I like to call it, `%NewMath%`.

**Review by kriskowal — CHANGES_REQUESTED** [review]
This is a breaking change to the extent that anyone relied on the old behavior. My feeling is that the old behavior sufficiently deterred relying upon the old behavior that this is not a breaking change. But, if you wish to play it safe, we can bump the 0.minor version and you’ll need a conventional commit to that effect.

Otherwise, this just needs an entry in `NEWS.md` under `# Next release` and an update to the docs, particularly the doc that outlines the differences in the Hardened JavaScript realm and compartment.

**Comment by erights** [comment]
> This is a breaking change to the extent that anyone relied on the old behavior. My feeling is that the old behavior sufficiently deterred relying upon the old behavior that this is not a breaking change. But, if you wish to play it safe, we can bump the 0.minor version and you’ll need a conventional commit to that effect.

Based on this, I just changed from `fix(ses)!:` to `fix(ses):`. Do I understand correctly that this is consistent with your answer above?

File: packages/ses/src/tame-date-constructor.js (hunk 3 above)
Comment (erights) [reply packages/ses/src/tame-date-constructor.js:88]: What about the big bang? Or the Unix Epoch?

**Comment by erights** [comment]
> Otherwise, this just needs an entry in `NEWS.md` under `# Next release` and an update to the docs, particularly the doc that outlines the differences in the Hardened JavaScript realm and compartment.

There are indeed stale *.md files that need updating. Working on it. By "docs", do you mean anything besides *.md files in the endo repo?

**Comment by kriskowal** [comment]
> > Otherwise, this just needs an entry in `NEWS.md` under `# Next release` and an update to the docs, particularly the doc that outlines the differences in the Hardened JavaScript realm and compartment.
> 
> There are indeed stale *.md files that need updating. Working on it. By "docs", do you mean anything besides *.md files in the endo repo?

Correct, just *.md in the ses package. I think just docs/reference.md and docs/guide.md. There are other stale docs that are out of scope.

**Comment by kriskowal** [comment]
> > This is a breaking change to the extent that anyone relied on the old behavior. My feeling is that the old behavior sufficiently deterred relying upon the old behavior that this is not a breaking change. But, if you wish to play it safe, we can bump the 0.minor version and you’ll need a conventional commit to that effect.
> 
> Based on this, I just changed from `fix(ses)!:` to `fix(ses):`. Do I understand correctly that this is consistent with your answer above?

Yes, we can treat this as non-breaking, unless you can think of a realistic working program that would break because of these changes.

**Comment by erights** [comment]
> Correct, just *.md in the ses package. I think just docs/reference.md and docs/guide.md. There are other stale docs that are out of scope.

Done. The second has these. A lot more than either of us expected. PTAL.

**Review by kriskowal — APPROVED** [review]

**Pull request merged**